### ISSUE TITLE
index camliContent pointing to a bytes

### ISSUE THREAD

**Issue by bradfitz** [camlistore/camlistore#42]
<pre>If a permanode has attribute &quot;camliContent&quot; pointing to a &quot;bytes&quot;
JSON schema, we should find the sha1 of the bytes content (which might be of unbounded
size), to prevent dup uploads in the future.

This is continuing work from <a href="https://camlistore.org/issue/17">issue #17</a> (file/bytes split).</pre>

**Comment by mpl** [comment]
https://camlistore-review.googlesource.com/4806 adds the camput feature. I've experimented a bit with it, and it looks like almost everything copes well with it. However, if you do
camput file -permanode --contents_only foo
and you then d&d foo in the web UI to upload it, no new file schema blob will be created. While one may have expected a new schema to be created since uploading through the UI should create a schema with the file's attributes, which would be different from the one we already created with camput. I'm not sure I would have considered that behaviour a bug, except that if you then do
camput file -permanode foo
then a new different file schema is created as expected. So maybe the UI should be capable of behaving that way as well.

On a related note, I'd really like to clean up files.go and move the Uploader to its own package.
